### PR TITLE
Move the language server out of the vscode plugin.

### PR DESCRIPTION
**Pull request by rictic**
- [x] CHANGELOG.md has been updated

Moved from https://github.com/polymer/vscode-plugin

Once this is released we'll migrate the other plugins to use this rather than our home-grown protocol, and deprecate then remove the editor server.

### CODE REVIEW

**Comment by FredKSchott** [comment]
TIL about https://github.com/Microsoft/language-server-protocol

**Comment by rictic** [comment]
Yes! I haven't been communicating about it enough, but I've got a design doc going out soon.

Thanks for the review!

File: package.json [hunk 1]
```diff
@@ -44,6 +44,7 @@
     "@types/split": "^0.3.28",
     "polymer-analyzer": "=2.0.0-alpha.16",
     "split": "^1.0.0",
-    "typescript": "^2.0.3"
+    "typescript": "^2.0.3",
+    "vscode-languageserver": "^3.0.0-alpha.1"
```
Comment (FredKSchott) [inline package.json]: pin to `=3.0.0-alpha.1`?

Comment (rictic) [inline package.json]: So far they've been pretty good about not breaking in alpha point releases, but will revisit if it becomes an issue.

**Comment by rictic** [comment]
Merged into master. Accidentally didn't update the `base` of this PR beforehand, and can't do it now, so.. closing.

**Pull request closed without merging**